cmd/libs/go2idl/openapi-gen/generators/openapi.go

```diff
@@ -362,26 +362,68 @@ func (g openAPITypeWriter) generate(t *types.Type) error {
 	return nil
 }
 
-func (g openAPITypeWriter) generateExtensions(CommentLines []string) error {
+// getTagsMap extract the key-value pairs from openapi tags and struct tags returns a map[string]string
+func getTagsMap(CommentLines []string, tags string) (map[string]string, error) {
+	tagMap := make(map[string]string)
+
+	// Handle openapi tags
 	tagValues := getOpenAPITagValue(CommentLines)
-	anyExtension := false
 	for _, val := range tagValues {
 		if strings.HasPrefix(val, tagExtensionPrefix) {
-			if !anyExtension {
-				g.Do("VendorExtensible: spec.VendorExtensible{\nExtensions: spec.Extensions{\n", nil)
-				anyExtension = true
-			}
 			parts := strings.SplitN(val, ":", 2)
 			if len(parts) != 2 {
-				return fmt.Errorf("Invalid extension value: %v", val)
+				return nil, fmt.Errorf("Invalid extension value: %v", val)
 			}
-			g.Do("\"$.$\": ", parts[0])
-			g.Do("\"$.$\",\n", parts[1])
+			tagMap[parts[0]] = parts[1]
 		}
 	}
-	if anyExtension {
-		g.Do("},\n},\n", nil)
+
+	// Handle struct tags
+	tagPairs := strings.Fields(tags)
+	for _, tagPair := range tagPairs {
+		parts := strings.SplitN(tagPair, ":", 2)
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("Invalid extension value: %v", tagPair)
+		}
+		switch parts[0] {
+		case "patchStrategy":
```

Comment:
Yes, every apiserver should support strategic merge patch strategies.  The code for the strategic merge patch exists under `k8s.io/apimachinery/pkg/util/strategicpatch` and it looks like it is wired up to `PATCH` correctly.  I have tested that it actually works correctly in aggregated API servers, but if it doesn't, it should.

If you prefer to use the +xxxx comment tag over the struct tags we could support that as well.  I talked with Daniel a bit, and he thinks we should support some notion of linting.  At least for this feature, would prefer to do the linting as part of the generation itself, instead of decoupling the 2.  WDYT of the following rollout plan:

1. Add open-api tags off of struct tag
1. Add comment tag to structs to duplicate struct tag patch information - incrementally do this per type
1. After adding comment tags to match all struct tags, validate that they match when doing the open-api generation